main.ts

```diff
@@ -257,13 +257,14 @@ export default class MyPlugin extends Plugin {
 				const end = e.position.end;
 				myLine.formatted = myLine.formatted.replace(line.substring(start.col, end.col), e.displayText);
 			});
+
+			// Regex of all the supported alternate checkbox styles
+			const cbRe = /^- \[[ x\/\-><?!*\"lbiSpcfkwud]\]/gi;
 			if (ignoreCheckboxes) {
-				if (myLine.formatted.trimLeft().startsWith("- [x]")) {
-					myLine.formatted = myLine.formatted.replace("- [x]", "");
-				}
+				myLine.formatted = myLine.formatted.replace(cbRe, "");
```

Comment:
What about indention? List items don't have to start at the beginning of the line.

Reply:
Good catch. Updated the regex to allow for optional white space characters at the beginning of the line before the start of the checkbox.